### PR TITLE
Use new adin driver in L2 and BCMP

### PR DESCRIPTION
**Pull request by towynlin**
This commit gets hello world to build, but other apps will need similar changes.

Pulls in https://github.com/bristlemouth/bm_core/pull/40 which should be merged first, then this PR updated with the `bm_core` merge commit before merging this one.

I wanted to get reviewer eyes on this before I go changing all the apps.

### CODE REVIEW

File: src/lib/bcmp/bm/bristlemouth.cpp [hunk 1]
```diff
@@ -61,35 +53,16 @@ void bcl_init(void) {
...
+  if (err != BmOK) {
+    printf("ADIN2111 init error %d\n", err);
+  }
 
+  bm_l2_init(network_device);
   bm_ip_init();
-
-  bcmp_init();
-
+  bcmp_init(network_device);
```
Comment (towynlin) [inline src/lib/bcmp/bm/bristlemouth.cpp:65]: Let's goooooooo! 🚀

File: src/lib/drivers/adin2111/src/adi_hal.c [hunk 2]
```diff
@@ -1,113 +1,21 @@
...
-
-#include <string.h>
 #include "adi_hal.h"
 #include "adi_bsp.h"
+#include "bsp.h"
+#include <string.h>
 
-/* Lookup table of CRC32 constants */
-static const uint32_t crc32_tab[] = {
-	0x00000000, 0x77073096, 0xEE0E612C, 0x990951BA, 0x076DC419, 0x706AF48F,
```
Comment (towynlin) [inline src/lib/drivers/adin2111/src/adi_hal.c:22]: This is in a separate adi_fcs.c in the bmcore adin driver. Other parts of this file are the locally required abstraction implementations though.

File: src/lib/bcmp/bm/bristlemouth_client.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      .ver_minor = minor,
+      .ver_patch = patch,
+      .sn = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'},
+  };
+
+  printf("Starting up BCL\n");
+  device_init(device);
+
+  BmErr err = bristlemouth_init(adin_power_callback);
+  if (err == BmOK) {
```
Comment (towynlin) [inline src/lib/bcmp/bm/bristlemouth_client.cpp:75]: LOOK AT THAT!!! 🕺  This is the entire code path between bm_protocol and bm_core!!!!!!

**Review by victorsowa12 — APPROVED** [review]
Looks great!

**Comment by towynlin** [comment]
OK, so CI is finally passing. I'm trying to run hello world on a dev kit and getting adin SPI errors during init. I'll debug that next week. ✌️

File: src/apps/bm_devkit/serial_bridge/CMakeLists.txt [hunk 4]
```diff
@@ -65,7 +65,6 @@ set(LIB_FILES
     ${SRC_DIR}/lib/debug/debug_bm_service.cpp
     ${SRC_DIR}/lib/debug/debug_configuration.cpp
     ${SRC_DIR}/lib/drivers/abstract/abstract_i2c.cpp
-    ${SRC_DIR}/lib/drivers/adin2111/src/adi_bsp.c
```
Comment (victorsowa12) [inline src/apps/bm_devkit/serial_bridge/CMakeLists.txt:68]: should the other adi_mac/phy/spi_oa files be removed like in the other apps cmakelists?

Should the BSP_IONCLUDES be included as well?

Same comment applies to the rx_live_example.

**Review by victorsowa12 — APPROVED** [review]
🤘

File: src/apps/bm_devkit/serial_bridge/CMakeLists.txt (hunk 4 above)
Comment (victorsowa12) [reply src/apps/bm_devkit/serial_bridge/CMakeLists.txt:68]: I believe @towynlin is already working on the rx live example 👍

File: src/apps/adin_test/CMakeLists.txt [hunk 5]
```diff
@@ -59,13 +59,7 @@ set(LIB_FILES
     ${SRC_DIR}/lib/debug/debug_bm_service.cpp
     ${SRC_DIR}/lib/debug/debug_configuration.cpp
 
-    ${SRC_DIR}/lib/drivers/adin2111/src/adi_bsp.c
     ${SRC_DIR}/lib/drivers/adin2111/src/adi_hal.c
-    ${SRC_DIR}/lib/drivers/adin2111/src/adi_mac.c
-    ${SRC_DIR}/lib/drivers/adin2111/src/adi_phy.c
-    ${SRC_DIR}/lib/drivers/adin2111/src/adi_spi_oa.c
-    ${SRC_DIR}/lib/drivers/adin2111/src/adin2111.c
-    ${SRC_DIR}/lib/drivers/adin2111/src/eth_adin2111.cpp
```
Comment (matt001k) [inline src/apps/adin_test/CMakeLists.txt:68]: I think we should remove these files from the repository as they now exist in bm_core.

File: src/apps/bm_devkit/serial_bridge/CMakeLists.txt (hunk 4 above)
Comment (towynlin) [reply src/apps/bm_devkit/serial_bridge/CMakeLists.txt:68]: Yep, and now fixed this one too. Thanks!

**Pull request merged**